src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/client/AssessRisksNeedsApiClient.kt

```diff
@@ -1,43 +1,51 @@
 package uk.gov.justice.digital.hmpps.hmppsworkload.client
 
+import kotlinx.coroutines.flow.toList
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Qualifier
-import org.springframework.core.ParameterizedTypeReference
-import org.springframework.http.HttpHeaders
 import org.springframework.stereotype.Component
 import org.springframework.web.reactive.function.client.WebClient
-import reactor.core.publisher.Mono
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import org.springframework.web.reactive.function.client.awaitBody
+import org.springframework.web.reactive.function.client.bodyToFlow
 import uk.gov.justice.digital.hmpps.hmppsworkload.client.dto.RiskPredictor
 import uk.gov.justice.digital.hmpps.hmppsworkload.client.dto.RiskSummary
-import java.util.Optional
 
 @Component
 class AssessRisksNeedsApiClient(@Qualifier("assessRiskNeedsApiWebClient") private val webClient: WebClient) {
 
-  fun getRiskSummary(crn: String, token: String): Mono<Optional<RiskSummary>> {
-    return webClient
-      .get()
-      .uri("/risks/crn/$crn/summary")
-      .header(HttpHeaders.AUTHORIZATION, token)
-      .retrieve()
-      .bodyToMono(RiskSummary::class.java)
-      .retry(1)
-      .map { Optional.of(it) }
-      .onErrorResume {
-        Mono.just(Optional.empty())
+  suspend fun getRiskSummary(crn: String): RiskSummary? {
+    repeat(2) {
```

Comment:
instead of the retry, this is 0 indexed so will repeat as 0,1 unless return is successful

Reply:
Hmm.. well I used retry in the lookup service - https://github.com/ministryofjustice/hmpps-staff-lookup-service/blob/9849e02723435050492a882d63fc5e81cbaac93d/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstafflookupservice/client/MicrosoftGraphClient.kt#L45.

Also, reading the `repeat` docs i don't see anything saying what it does on error. I don't think it tries again